ML project/Customer_churn_prediction.py

```diff
@@ -0,0 +1,512 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+get_ipython().run_line_magic('matplotlib', 'inline')
+
+
+# In[2]:
+
+
+df = pd.read_csv('train.csv')
+
+
+# In[3]:
+
+
+df.head()
+
+
+# In[4]:
+
+
+df.info()
+
+
+# In[5]:
+
+
+df.var()
+
+
+# Looks like 4 columns have high variance and would possibly benefit some kind of normalization
+
+# In[6]:
+
+
+sns.heatmap(df.corr(),cmap='BuGn');
+
+
+# Looks like there is multicollinearity among certain numerical features, hence some will be dropped
+
+# In[7]:
+
+
+df = df.copy()
+df.drop(['total_day_charge','total_eve_charge','total_night_charge','total_intl_charge'],axis=1,inplace=True)
+
+
+# In[8]:
+
+
+df.head()
+
+
+# In[9]:
+
+
+df.shape
+
+
+# In[10]:
+
+
+df[df.select_dtypes(['int64','float64']).columns].boxplot()
+plt.xticks(rotation=90);
+
+
+# The boxplot overview suggests that, since some outliers are in, the options are either proceed with the removal or adopt a robust scaling. Let's try out removal
+
+# In[11]:
+
+
+cols= list(df.select_dtypes(['int64','float64']).columns)
+
+upper_limit = df[cols].quantile(0.75) + 1.5*(df[cols].quantile(0.75)-df[cols].quantile(0.25))
+lower_limit = df[cols].quantile(0.25) - 1.5*(df[cols].quantile(0.75)-df[cols].quantile(0.25))
+cleaned_df = df[~((df[cols]>upper_limit)|(df[cols]<lower_limit)).any(axis=1)]
+cleaned_df.shape
```

Comment:
@Ettore1991 using `scipy` and zscore you can have code more compact

```
from scipy import stats
df[(np.abs(stats.zscore(df)) < 3).all(axis=1)]
```

Link to stackoverflow with example and detail: https://stackoverflow.com/a/23202269